packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,993 @@
+}
+
+/**
+ * Properties used to configure a global table
+ */
+export interface GlobalTableProps extends TableOptionsV2, SchemaOptions {
+  /**
+   * The name of all replica tables in the global table.
+   *
+   * @default - generated by CloudFormation
+   */
+  readonly tableName?: string;
+
+  /**
+   * The name of the TTL attribute for all replica tables in the global table.
+   *
+   * @default - TTL is disabled
+   */
+  readonly timeToLiveAttribute?: string;
+
+  /**
+   * The removal policy applied to the global table.
+   *
+   * @default RemovalPolicy.RETAIN
+   */
+  readonly removalPolicy?: RemovalPolicy;
+
+  /**
+   * The billing mode and the associated read and write capacity settings for all replica
+   * tables in the global table.
+   *
+   * Note: Read capacity is configurable on a per-replica basis.
+   *
+   * @default Billing.onDemand()
+   */
+  readonly billing?: Billing;
+
+  /**
+   * Properties used to configure the replica tables in the global table.
+   *
+   * Note: You can add or remove replicas after table creation, but you can only add or
+   * remove a single replica in each update.
+   *
+   * @default - a single replica table will be created in the region the global table is
+   * deployed to
+   */
+  readonly replicas?: ReplicaTableProps[];
+
+  /**
+   * Global secondary indexes to define on all replica tables in the global table.
+   *
+   * Note: Global secondary indexes can be configured on a per-replica basis. Tables only
+   * support a maximum of 20 global secondary indexes. You can only create or delete one
+   * global secondary index in a single stack operation.
+   *
+   * @default - no global secondary indexes
+   */
+  readonly globalSecondaryIndexes?: GlobalSecondaryIndexPropsV2[];
+
+  /**
+   * Local secondary indexes to define on all replica tables in the global table.
+   *
+   * Note: Tables only support a maximum of 5 local secondary indexes.
+   *
+   * @default - no local secondary indexes
+   */
+  readonly localSecondaryIndexes?: LocalSecondaryIndexProps[];
+
+  /**
+   * The server-side encryption to apply to all replica tables in the global table.
+   *
+   * Note: If the encryption type is customer managed, you must provide the ARN of a KMS
+   * key for each replica region.
+   *
+   * @default TableEncryptionV2.dynamoOwnedKey()
+   */
+  readonly encryption?: TableEncryptionV2;
+}
+
+/**
+ * Represents an instance of a global table
+ */
+export interface IGlobalTable extends IResource {
+  readonly tableArn: string;
+  readonly tableName: string;
+  readonly tableId?: string;
+  readonly tableStreamArn?: string;
+}
+
+/**
+ * Attributes of a global table
+ */
+export interface GlobalTableAttributes {
+  readonly tableArn?: string;
+  readonly tableName?: string;
+  readonly tableId?: string;
+  readonly tableStreamArn?: string;
+}
+
+/**
+ * The base class for a global table
+ */
+abstract class GlobalTableBase extends Resource implements IGlobalTable {
```

Comment:
Looks like this class ain't doing much. But I suppose this is where a lot of the `ITable` implementation will go in the future, yeah?

Reply:
Correct. ITable implementation has been added.